src/nl_solvers/newtons_method.jl

```diff
@@ -423,7 +423,7 @@ function allocate_cache(alg::KrylovMethod, x_prototype)
         jacobian_free_jvp_cache = isnothing(jacobian_free_jvp) ? nothing :
                                   allocate_cache(jacobian_free_jvp, x_prototype),
         forcing_term_cache = allocate_cache(forcing_term, x_prototype),
-        solver = type(l, l, args..., Krylov.ktypeof(x_prototype); kwargs...),
+        solver = type(l, l, args..., ClimaComms.array_type(x_prototype); kwargs...),
```

Comment:
From the error message in CI, it looks like we should be passing the context here?